sdk/src/main/java/com/google/cloud/dataflow/sdk/util/TimerInternals.java

```diff
@@ -204,9 +205,21 @@ public String toString() {
           .toString();
     }
 
+    /**
+     * {@inheritDoc}.
+     *
+     * <p>The ordering of {@link TimerData} that are not in the same namespace or domain is
+     * arbitrary.
+     */
     @Override
     public int compareTo(TimerData o) {
-      return Long.compare(timestamp.getMillis(), o.getTimestamp().getMillis());
+      ComparisonChain chain = ComparisonChain.start()
+          .compare(timestamp.getMillis(), o.getTimestamp().getMillis())
+          .compare(domain, o.domain);
+      if (chain.result() == 0) {
```

Comment:
I'm guessing this is being done because `namespace.stringKey()` may be expensive. If so comment on that, or else someone will suggest (or just clean this up) by suggesting getting rid of the `if`.